src/network/networkd-link.c

```diff
@@ -1432,6 +1432,42 @@ int link_set_mtu(Link *link, uint32_t mtu, bool force) {
         return 0;
 }
 
+static bool link_reduces_vlan_mtu(Link *link) {
+        /* See netif_reduces_vlan_mtu() in kernel. */
+        return streq_ptr(link->kind, "macsec");
+}
+
+static uint32_t link_get_stacked_netdevs_mtu(Link *link) {
+        uint32_t mtu = 0;
+        NetDev *dev;
+        Iterator i;
+
+        HASHMAP_FOREACH(dev, link->network->stacked_netdevs, i)
+                if (dev->kind == NETDEV_KIND_VLAN && dev->mtu > 0)
+                        /* See vlan_dev_change_mtu() in kernel. */
+                        mtu = MAX(mtu, link_reduces_vlan_mtu(link) ? dev->mtu + 4 : dev->mtu);
+
```

Comment:
Why we are bumping up the MTU. Should not the user responsible for this ?

Reply:
Otherwise, vlan devices may not work... Of course, just warning about that may be ok. But, when we know what is a minimal requirement, then I think it may helpful to bump MTU.

Reply:
And if we do not bump MTU, then user may not find why vlan does not work. See #5972.

Reply:
For that we just need to log it and should not increase.